checker/tests/initialization/EISOPIssue363a.java

```diff
@@ -0,0 +1,92 @@
+// Test case for issue 363:
+// https://github.com/eisop/checker-framework/issues/363
+
+import org.checkerframework.checker.initialization.qual.Initialized;
+import org.checkerframework.checker.initialization.qual.NotOnlyInitialized;
+import org.checkerframework.checker.initialization.qual.UnknownInitialization;
+
+class MyException extends Exception {
+    @NotOnlyInitialized EISOPIssue363a cause;
+
+    MyException(EISOPIssue363a cause) {
+        this.cause = cause;
+    }
+
+    MyException(@Initialized EISOPIssue363a cause, String dummy) {
+        this.cause = cause;
+    }
+
+    MyException(@UnknownInitialization EISOPIssue363a cause, int dummy) {
+        this.cause = cause;
+    }
+}
+
+public class EISOPIssue363a {
+    Object field;
+
+    EISOPIssue363a() throws MyException {
+        // :: error: (throw.type.incompatible)
+        // :: error: (argument.type.incompatible)
+        throw new MyException(this);
+    }
+
+    EISOPIssue363a(boolean dummy) throws MyException {
+        // :: error: (throw.type.incompatible)
+        throw new MyException(this, 0);
+    }
+
+    EISOPIssue363a(char dummy) throws MyException {
```

Comment:
It's rather misleading that this constructor, with a `char` parameter, invokes the `MyException` constructor with a String argument. The following constructor with a `String` parameter, invokes a different constructor. Maybe these can be aligned a bit better?